src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -0,0 +1,263 @@
+                                               final Predicate<VariantAnnotation> addAnnot) {
+        if ( genotypeAnnotations.isEmpty() ) {
+            return vc.getGenotypes();
+        }
+
+        final GenotypesContext genotypes = GenotypesContext.create(vc.getNSamples());
+        for ( final Genotype genotype : vc.getGenotypes() ) {
+            PerReadAlleleLikelihoodMap perReadAlleleLikelihoodMap = null;
+            if (stratifiedPerReadAlleleLikelihoodMap != null) {
+                perReadAlleleLikelihoodMap = stratifiedPerReadAlleleLikelihoodMap.get(genotype.getSampleName());
+            }
+
+            final GenotypeBuilder gb = new GenotypeBuilder(genotype);
+            for ( final GenotypeAnnotation annotation : genotypeAnnotations) {
+                if (addAnnot.test(annotation)) {
+                    annotation.annotate(ref, vc, genotype, gb, perReadAlleleLikelihoodMap);
+                }
+            }
+            genotypes.add(gb.make());
+        }
+
+        return genotypes;
+    }
+
+    private static final class AnnotationManager {
+
+        private final List<String> annotationGroupsToUse;
+        private final List<String> annotationsToUse;
+
+        private AnnotationManager(final List<String> annotationGroupsToUse, final List<String> annotationsToUse){
+            this.annotationGroupsToUse = annotationGroupsToUse;
+            this.annotationsToUse = annotationsToUse;
+
+            final Set<String> unknownAnnots = Sets.difference(new HashSet<>(annotationsToUse), new HashSet<>(AnnotationManager.getAllAnnotationNames()));
+            if (!unknownAnnots.isEmpty()){
+                throw new BadArgumentValue("annotation", "Annotations " + unknownAnnots + " were not found; please check that you have specified the annotation name correctly");
+            }
+
+            final Set<String> unknownGroups =  Sets.difference(new HashSet<>(annotationGroupsToUse), new HashSet<>(AnnotationManager.getAllAnnotationGroupNames()));
+            if (!unknownGroups.isEmpty()){
+                throw new BadArgumentValue("group", "Unknown annotation group in " + unknownGroups + ". Known groups are " + Arrays.toString(AnnotationGroup.values()) );
+            }
+
+        }
+
+        static AnnotationManager ofSelected(final List<String> annotationGroupsToUse, final List<String> annotationsToUse, final List<String> annotationsToExclude){
+            final List<String> groups = new ArrayList<>(annotationGroupsToUse);//make copy
+            final List<String> annots = new ArrayList<>(annotationsToUse);//make copy
+            annots.removeAll(annotationsToExclude);
+            return new AnnotationManager(groups, annots);
+        }
+
+        static AnnotationManager ofAll(final List<String> annotationsToExclude){
+            final List<String> groups = getAllAnnotationGroupNames();
+            final List<String> annots = getAllAnnotationNames();
+            annots.removeAll(annotationsToExclude);
```

Comment:
BUG: if an annotation is specifically excluded, but its group is included, it gets included despite being individually blacklisted. Recommended solution is to process `annotationsToExclude` at the `filterAnnotations` level.